paimon-core/src/main/java/org/apache/paimon/operation/LocalOrphanFilesClean.java

```diff
@@ -127,10 +134,30 @@ public CleanOrphanFilesResult clean()
                         .collect(Collectors.toList()));
         candidateDeletes.clear();
 
+        // clean empty directory
+        if (!dryRun) {
+            cleanEmptyDirectory(deleteFiles);
+        }
+
         return new CleanOrphanFilesResult(
                 deleteFiles.size(), deletedFilesLenInBytes.get(), deleteFiles);
     }
 
+    private void cleanEmptyDirectory(List<Path> deleteFiles) {
```

Comment:
1. Extract a common method that can be used by both core, spark and so on. Note: bucket and partitions logic can be processed together, as long as we set level = partitionKeysNum + 1.
2. Let the entire method be executed through randomlyOnlyExecute.